### PR TITLE
Blog post: Patterns for project failure

### PR DESCRIPTION
**Pull request by njuneau-coveo**

### CODE REVIEW

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  [error budgets](https://www.atlassian.com/incident-management/kpis/error-budget) anymore.
+
+## Step 8: Security?
+
+If there is some time left, security measures may be put in place on the off chance that they’re not
+already there. Security is a lot like UX: it’s best done separately, fixed most efficiently once in
+production. Your well-spent budget allows for the following security considerations to be put in
+place:
+
+*
```
Comment (larosek) [inline _posts/2023-09-14-patterns-for-project-failure.md:210]: Seems like this should not be here no?

**Review by larosek — APPROVED** [review]
Don't forget to change the date :D
Good job! I chuckle a few times hehe

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 1 above)
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md:210]: The paragraph is voluntarily left blank - I'll add a hint that the project didn't have any budget left 😆

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+to name a few. However, deploying those tools efficiently requires a lot of effort. If you don’t
+have time to automate your infrastructure now, you won’t have time later either. Just like UX,
+implementing infrastructure automation late in the game costs much less than doing it in advance.
+After all, what’s not to love about pages of AWS Web console screenshots,
+<abbr title="Bourne-again shell">Bash</abbr> scripts and good old elbow grease?
+
+## Step 7: There are no 9s after 100%
+
+You made it from steps 1 to 6 and your product has been deployed in production. People are using
+your product and the future looks bright. However, you want to ensure that _PugBuddies_ remains
```
Comment (amoreauCoveo) [inline _posts/2023-09-14-patterns-for-project-failure.md:184]: Wait is it a joke that the product keeps changing names? 😛

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+eternally so: functioning all the time, in tip-top shape.
+
+This is where one sneaky failure pattern rears its head: being unable to fail. You want your product
+to be perfect, and you know you can make it:
+
+* Simple <abbr title="Service-level objective">SLOs</abbr>: Always on, all the time. Pug love is
```
Comment (amoreauCoveo) [inline _posts/2023-09-14-patterns-for-project-failure.md:190]: always should be all lowercase

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+  eternal and cannot fail. [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem) be damned, your
+  software will scale,
+  distribute and remain perfectly consistent forever.
```
Comment (amoreauCoveo) [inline _posts/2023-09-14-patterns-for-project-failure.md:193]: distribute, and remain (oxford comma)

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+already there. Security is a lot like UX: it’s best done separately, fixed most efficiently once in
+production. Your well-spent budget allows for the following security considerations to be put in
+place:
+
+* 
+
+
+## Epilogue
+
+By now _PugVogue_ should be up in production, with a team happily maintaining it day to day in a
```
Comment (amoreauCoveo) [inline _posts/2023-09-14-patterns-for-project-failure.md:215]: okay I get the joke now

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+healthy manner. It will stand as a shining beacon of innovation inside your organization, a
+testament to your everlasting creativity. Please excuse me, my pager is buzzing!
+
+## Conclusion
+
+I would hope everybody recognizes themselves in one way or another while going through this article.
+Whether it is your past self you are seeing, the self you wish to improve or a self you wish not to
```
Comment (amoreauCoveo) [inline _posts/2023-09-14-patterns-for-project-failure.md:222]: the self you wish to improve, or a self . . . (oxford comma)

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 1 above)
Comment (amoreauCoveo) [reply _posts/2023-09-14-patterns-for-project-failure.md:210]: Leaving an empty bullet point doesn't look great: https://njuneau-coveo.github.io/source.coveo.com/2023/09/14/patterns-for-project-failure/

**Review by amoreauCoveo — COMMENTED** [review]
gg!

**Review by amoreauCoveo — APPROVED** [review]

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+
+* Defining software requirements, especially with frameworks such as
+  [FURPS+](https://web.archive.org/web/20201112020231/http://www.ibm.com/developerworks/rational/library/4706.html#N100A7)
+  or worse, [ISO/IEC 25010:2011](https://www.iso.org/standard/35733.html) is unnecessary. All people
+  think alike and share the same perspective about the
+  project. Having a standard on how to communicate requirements is just another burden.
+* Time spent drafting software processes, especially with well-known languages such as
+  [UML](https://www.uml.org/) or [BPMN](https://www.bpmn.org/), is time not spent coding. Besides,
+  UX does not have access to those artifacts and just like UX,
+  the most cost-effective way to tackle core issues is once the code is being built.
```
[the remark this answers is not in the document]
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md:140]: Actually, it's more `while the code is being built` - wdyt?

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 2 above)
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md:184]: It is :) - it's another (implicit) sign of a project that goes well: changing its name constantly :P

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 3 above)
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md:190]: Service-level objectices are commonly abbreviated "SLOs" 🤔 https://en.wikipedia.org/wiki/Service-level_objective

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 6 above)
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: Fixed

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 1 above)
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: Fixed paragraph to address confusiom let me know what you think!

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 4 above)
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: Fixed

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 3 above)
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: Fixed

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+neat thing about open source software is that you can make it your own. Key benefits include:
+
+* Internal forking: no need to spend time trying to get your fix upstream, just ship it internally.
+* Reduced license costs: Community support ended and you don’t pay for commercial support? Keep the
+  patches coming yourself!
+
+## Step 6: Automate infrastructure work later
+
+Many tools exist to automate infrastructure work: [Terraform](https://www.terraform.io/),
+[Cloudformation](https://docs.aws.amazon.com/cloudformation/) or [Ansible](https://www.ansible.com/)
```
[the remark this answers is not in the document]
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: Fixed

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+  [UML](https://www.uml.org/) or [BPMN](https://www.bpmn.org/), is time not spent coding. Besides,
+  UX does not have access to those artifacts and just like UX,
+  the most cost-effective way to tackle core issues is once the code is being built.
+* Keeping an Architecture Decision Record (ADR) is time-consuming. The context in which the
+  decisions are made never changes and will never invalidate your decisions in the future. You’ll
+  save a lot of time revisiting decisions you should not question ever anyway. If you’re onboarding
+  new employees that have a hard time getting up to speed as a result, it’s their fault they weren’t
+  there when the decisions were made.
+
+In short: get to code (i.e. _real_ value) as fast as you can.
```
[the remark this answers is not in the document]
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: Fixed

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+
+## Introduction
+
+My name is Nicolas Juneau and I am Coveo’s _CFO_ (Chief Failure Officer). As the blog review team
+has yet to unpack a huge backlog of articles to review, I took this opportunity to write on this
+blog about a subject nobody wants me to talk about: patterns for project failure.
+
+We all heard the conferences, we all read the articles: we know how to ensure a project’s success.
+After all, software engineering is a tried and true discipline as old as civil engineering. Julius
+Caesar successfully designed, wrote and deployed
```
[the remark this answers is not in the document]
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: Fixed

File: _posts/2023-09-14-patterns-for-project-failure.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+As a Coveo employee, a colleague came to you with the following project opportunity:
+
+To build and deploy a social network tailor-made for Pug lovers. Not the toy, but arguably the most
+lovable dog. For brevity’s sake, sales and marketing already came up with the name: _PikPug_.
+
+The details of the project are intentionally left vague.
+
+## Step 1: Business case (non)-analysis
+
+_PugBook_ is bound to be a surefire hit: market research shows Pug lovers are all over the place and
```
[the remark this answers is not in the document]
Comment (njuneau-coveo) [reply _posts/2023-09-14-patterns-for-project-failure.md:57]: It's a surprise!

File: _posts/2023-09-14-patterns-for-project-failure.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (amoreauCoveo) [reply _posts/2023-09-14-patterns-for-project-failure.md]: that's good :)

**Pull request merged**